### PR TITLE
[Task 1862938] implemented storage azure service.

### PR DESCRIPTION
**Pull request by shenqianjin**
[AB#1862938](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1862938)

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    private AzureStorageAccount(@Nonnull final List<Subscription> subscriptions) {
+        super(AzureStorageAccount::new, subscriptions);
+    }
+
+    public List<StorageAccount> list() {
+        return getSubscriptions().stream()
+            .map(subscription -> StorageManagerFactory.create(subscription.getId()))
+            .flatMap(manager -> manager.storageAccounts().list().stream())
+            .collect(Collectors.toList()).stream()
+            .map(a -> new StorageAccount(a.manager(), a))
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java:52]: rename a to a meanful name

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    private AzureStorageAccount(@Nonnull final List<Subscription> subscriptions) {
+        super(AzureStorageAccount::new, subscriptions);
+    }
+
+    public List<StorageAccount> list() {
+        return getSubscriptions().stream()
+            .map(subscription -> StorageManagerFactory.create(subscription.getId()))
+            .flatMap(manager -> manager.storageAccounts().list().stream())
+            .collect(Collectors.toList()).stream()
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java:51]: remove this line

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java (hunk 1 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java]: renamed.

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java (hunk 2 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java]: removed.

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        final com.azure.resourcemanager.storage.models.StorageAccount account =
+                StorageManagerFactory.create(getDefaultSubscription().getId()).storageAccounts().getByResourceGroup(resourceGroup, name);
+        return new StorageAccount(account.manager(), account);
+    }
+
+    public CheckNameAvailabilityResultEntity checkNameAvailability(String subscriptionId, String name) {
+        StorageManager manager = StorageManagerFactory.create(subscriptionId);
+        CheckNameAvailabilityResult result = manager.storageAccounts().checkNameAvailability(name);
+        return new CheckNameAvailabilityResultEntity(result.isAvailable(),
+                Optional.ofNullable(result.reason()).map(Reason::toString).orElse(null), result.message());
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java:69]: Optional.ofNullable(result.reason()).map(Reason::toString).orElse(null) -> Objects.toString(result.reason())

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/StorageManagerFactory.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import com.microsoft.azure.toolkit.lib.AzureConfiguration;
+import com.microsoft.azure.toolkit.lib.auth.Account;
+import com.microsoft.azure.toolkit.lib.auth.AzureAccount;
+import com.microsoft.azure.toolkit.lib.common.cache.Cacheable;
+
+import java.util.Optional;
+
+public final class StorageManagerFactory {
+
+    private StorageManagerFactory() {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/StorageManagerFactory.java:22]: shall we remove the public in this class?

Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/StorageManagerFactory.java:22]: no need.

File: azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java (hunk 3 above)
Comment (shenqianjin) [reply azure-toolkit-libs/azure-toolkit-storage-lib/src/main/java/com/microsoft/azure/toolkit/lib/storage/service/AzureStorageAccount.java:69]: result may be null.

**Review by andxu — APPROVED** [review]

**Pull request merged**